tests/utils.py

```diff
@@ -189,10 +189,26 @@ def override_global_config(values):
             setattr(asm_config, key, value)
     # If ddtrace.settings.asm.config has changed, check _asm_can_be_enabled again
     asm_config._eval_asm_can_be_enabled()
+    from ddtrace.appsec._processor import AppSecSpanProcessor
+
+    AppSecSpanProcessor.disable()
     if asm_config._asm_enabled:
         from ddtrace.appsec._listeners import load_appsec
 
         load_appsec()
+    else:
+        if asm_config._api_security_active:
+            from ddtrace.appsec._api_security.api_manager import APIManager
+
+            APIManager.disable()
+    if asm_config._iast_enabled:
+        from ddtrace.appsec._iast.processor import AppSecIastSpanProcessor
+
+        AppSecIastSpanProcessor.enable()
+    else:
+        from ddtrace.appsec._iast.processor import AppSecIastSpanProcessor
+
+        AppSecIastSpanProcessor.disable()
```

Comment:
```suggestion
    from ddtrace.appsec._iast.processor import AppSecIastSpanProcessor
    
    if asm_config._iast_enabled:        
        AppSecIastSpanProcessor.enable()
    else:
        AppSecIastSpanProcessor.disable()
```

it's going to get loaded no matter what